src/main/java/com/coveo/feign/ReflectionErrorDecoder.java

```diff
@@ -244,11 +247,13 @@ private void extractExceptionInfo(Class<? extends S> clazz)
                 existingExceptionDetails.getClazz().getName()));
       }
 
-      if (!exceptionMessageHandlingLogged
+      if (detailMessageField == null
+          && !exceptionMessageHandlingLogged
           && !ExceptionMessageSetter.class.isAssignableFrom(clazz)) {
```

Comment:
should be BaseExceptionClass here no?

Reply:
No, this message is meant to warn the user when his baseExceptionClass does not implement `ExceptionMessageSetter` and that the reflection trick is not available